src/bundle.c

```diff
@@ -424,12 +422,16 @@ static int install_bundles(struct list *bundles, struct list **subs, int current
 	/* step 1: check bundle args are valid if so populate subs struct */
 	ret = add_subscriptions(bundles, subs, current_version, mom, 0);
 
-	if (ret) {
-		if (ret == 1) {
-			printf("bundle(s) already installed, exiting now\n");
-		}
+	switch (ret)
+	{
+	case 0: /* no errors, but nothing added */
+		printf("bundle(s) already installed, exiting now\n");
+		/* fall through */
+	case add_sub_ERR: /* error of some kind */
+	case add_sub_ERR|add_sub_NEW: /* error of some kind but we did add something */
 		ret = EBUNDLE_INSTALL;
 		goto out;
+	default: break;
```

Comment:
Ah goodness yes, I was something thinking of ret | (add_sub_ERR|add_sub_NEW) not a case statement matching exactly add_sub_ERR|add_sub_NEW X(.